src/cosalib/qemuvariants.py

```diff
@@ -251,10 +250,6 @@ def mutate_image(self):
                               f" expected: '{self.image_format}'"
                               f" found: '{img_info.get('format')}'"))
 
-        if self.mutate_callback:
-            log.info("Processing work image callback")
-            meta_patch.update(self.mutate_callback(work_img) or {})
```

Comment:
VMware still uses this.

Reply:
Ah, good catch...I was grepping for `mutate-callback` and not `mutate_callback`.

Reply:
Looks like this is still being removed, just in the first commit instead of the third.

Reply:
I removed the *declarative* interface, but not the actual invocation of the callback.

This code is trying to be OOP, so giving subclasses a hook makes total sense.  Having a "superclass/generic" declaration reference code from a "subclass" in the digitalocean case was both mixing declarative/imperative and violating the abstraction layer, so let's not encourage that again.

Reply:
Yup, I understand your intention, but check the diff.  The invocation has also been removed.

Reply:
Ooh yes.  Thanks for spotting that, it sailed through CI since nothing is checking the vmware ova is working.  OK at this point I'm just going to cut my losses and remove that commit and we can clean up the mutate stuff later.